### PR TITLE
Add Waiter for Athena

### PR DESCRIPTION
**Pull request by mag-chang**
I want waiter for Athena.
Because this issue(boto/boto3#1212) and this Pull Request(#1358) remained unattended, I was created a new Pull Request.

### CODE REVIEW

**Comment by Nop0x** [comment]
This is really needed as we always need to wait for the queries to complete!

**Comment by waynetaylor** [comment]
Any update on when this will be added?

File: botocore/data/athena/2017-05-18/waiters-2.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        },
+        {
+          "matcher": "path",
+          "expected": "FAILED",
+          "argument": "QueryExecution.Status.State",
+          "state": "failure"
+        },
+        {
+          "matcher": "path",
+          "expected": "CANCELED",
```
Comment (maciejmajewski) [inline botocore/data/athena/2017-05-18/waiters-2.json:23]: FYI this is a typo
```suggestion
          "expected": "CANCELLED",
```

**Comment by stobrien89** [comment]
Hi all,

Thanks again for the request. We are now submitting requests for paginators and waiters upstream so that all AWS SDKs benefit from a consistent implementation. I will pass on the request for these waiters to the service team. We will also update the guidance in the contributing guide to better reflect this policy. I've added an issue to track this here:

aws/aws-sdk#80

Thanks again for bringing this to our attention!

**Pull request closed without merging**